### PR TITLE
Refactor IIASA-connection-API and rework all related tests

### LINKED ISSUE #303: Add better documentation and testing of objects returned IIASA-db connection
We need to add better documentation and more unit tests of the objects returned via the Rest API of the IIASA database infrastructure.

See https://github.com/iiasa/ipcc_sr15_scenario_analysis/issues/28 for an issue where this caused problems for a user.

### LINKED ISSUE #378: Return-type of `iiasa.Connection.query()` not IamDataFrame
Working on some improvements to the integration of the IIASA-database-integration, and I'm wondering why the return-type `iiasa.Connection.query()` is a `pandas.DataFrame`, not an `IamDataFrame` - which would be more logical.

@gidden, any particular reason? And any objection to casting the returned dataframe as `IamDataFrame` directly (instead of having to do `df = pyam.IamDataFrame(iiasa.Connection(<name>).query(...))`)?

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added
# Description of PR

This PR harmonizes the behavior of the iiasa.Connection class functions, prepares for a refactoring of the IIASA-backend API, and reworks the tests to use a dedicated test database instance (making the tests more precise by comparing full `IamDataFrame` instance rather than just length of returned objects).

Changes In detail:
 - all iiasa-related tests are directed to a dedicated `integration-test` instance that has only 4 scenario objects based on the default objects used `conftest.py`, closes #303
- `read_iiasa()` and `query()` return an `IamDataFrame` instead of a pandas object
  (now including meta-indicators by default), closes #378 
- deprecate `scenario_list()` for an overview of all scenarios in favour of `index()`
- deprecate `available_metadata()` in favour of `meta_columns` (attribute)
- deprecate `metadata()` in favour of `meta()`

### CODE REVIEW

**Comment by danielhuppmann** [comment]
@stickler-ci is incorrectly reporting linting errors (they have been fixed)

**Comment by danielhuppmann** [comment]
@znicholls @byersiiasa @Rlamboll @gidden, FYI - this PR will make the integration with the IIASA-database-API more stable (because of more and more explicit tests), but it also adds one important change: the functions `read_iiasa()` and `query()` directly return an `IamDataFrame`. Other functions are marked as deprecated (including a log message), but will continue to work for at least one more release.

Apologies for the inconvenience, but I didn't find a practical way to avoid the immediate pain on the return type. Please ping any users that you know to be using this feature.

File: doc/source/tutorials/iiasa_dbs.ipynb [hunk 1]
```diff
@@ -76,7 +83,8 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We also provide some convenience functions to shorten the amount of code you have to write. Under the hood, `read_iiasa()` is just opening an connection to a database and making a query on that data.\n",
+    "We also provide some convenience functions to shorten the amount of code you have to write. Under the hood, `read_iiasa()` is just opening an connection to a database API and sends a query to the resource.\n",
```
Comment (znicholls) [inline doc/source/tutorials/iiasa_dbs.ipynb:86]:
```suggestion
    "We also provide some convenience functions to shorten the amount of code you have to write. Under the hood, `read_iiasa()` is just opening a connection to a database API and sends a query to the resource.\n",
```

Comment (danielhuppmann) [reply doc/source/tutorials/iiasa_dbs.ipynb]: thanks @znicholls!

**Pull request merged**